src/Features/ERDDAP/Platform/Observations/CurrentConditions/wind.tsx

```diff
@@ -1,81 +1,163 @@
 /**
  * Wind specific current conditions card
  */
-import React from "react"
-import { Card, CardBody, CardHeader } from "reactstrap"
+import * as React from "react"
+import { Link } from "react-router-dom"
+import { Card, CardBody, CardHeader, Col } from "reactstrap"
 
 import { round } from "Shared/math"
 import { DataTimeSeries } from "Shared/timeSeries"
 import { compassDirection } from "Shared/unitConversion"
+import { useUnitSystem } from "Features/Units"
 import { UnitSystem } from "Features/Units/types"
 import { converter } from "Features/Units/Converter"
 
 import { WindTimeSeriesChart } from "components/Charts"
 
-import { PlatformDataset } from "../../../types"
+import { UseDatasets } from "../../../hooks"
+import { PlatformFeature, PlatformTimeSeries } from "../../../types"
+import { pickWindDatasets, pickWindTimeSeries } from "../../../utils/wind"
+import { cardProps, observationLink } from "./common_card"
 
-interface Props {
-  datasets: PlatformDataset[]
-  unit_system: UnitSystem
+interface WindCardProps {
+  platform: PlatformFeature
 }
 
-export const WindCard: React.SFC<Props> = ({ datasets, unit_system }) => {
-  if (datasets.length < 1) {
-    return null
+/**
+ * If there are wind time series for a platform load and display recent data.
+ *
+ * @param platform Platform GeoJSON feature to display wind data from
+ */
+export const WindCard: React.FunctionComponent<WindCardProps> = ({ platform }) => {
+  const unit_system = useUnitSystem()
+
+  const aDayAgo = new Date()
+  aDayAgo.setDate(aDayAgo.getDate() - 1)
+
+  const { windTimeSeries, timeSeries } = pickWindTimeSeries(platform, aDayAgo)
+
+  // No current data or wind data don't exist for the platform
+  if (windTimeSeries.length < 1) {
+    const noWindTimeSeries = platform.properties.readings.filter((timeSeries) =>
+      timeSeries.data_type.standard_name.includes("wind")
+    )
+
+    if (noWindTimeSeries.length < 1) {
+      return null
+    }
+
+    return <OtherWindCard platform={platform}>No recent wind data available.</OtherWindCard>
   }
 
-  const data: DataTimeSeries[] = datasets.map((dataset) => ({
-    name: dataset.data_type.long_name,
-    timeSeries: dataset.readings,
-    unit: dataset.data_type.units,
-  }))
+  if (timeSeries.length < 0) {
+    return <OtherWindCard platform={platform}>No wind data</OtherWindCard>
+  }
 
-  const speed = datasets.filter((dataset) => dataset.data_type.standard_name.includes("speed"))
-  const gust = datasets.filter((dataset) => dataset.data_type.standard_name.includes("gust"))
-  const direction = datasets.filter((dataset) => dataset.data_type.standard_name.includes("direction"))
+  return (
+    <UseDatasets
+      timeSeries={timeSeries}
+      loading={<OtherWindCard platform={platform}>Loading wind data</OtherWindCard>}
```

Comment:
Codacy found an issue: ['OtherWindCard' was used before it was defined.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)